src/util/address.rs

```diff
@@ -522,6 +523,50 @@ impl Address {
         }
     }
 
+    /// Create a pay to taproot address from an untweaked key
+    /// 
+    /// This is done by using the equation Q = P + H(P|c)G, where 
+    ///  * Q is the tweaked key
+    ///  * P is the internal key
+    ///  * H is the hash function
+    ///  * c is the commitment data
+    ///  * G is the generator point
+    pub fn dangerous_p2tr<C: secp256k1::Verification>(
+        secp: Secp256k1<C>, 
+        internal_key: schnorrsig::PublicKey, 
+        merkle_root: Option<TapBranchHash>, 
+        network: Network
+    ) -> Result<Address, Error> {
+        //Commitment data.
+        let h: Vec<u8> = match merkle_root {
+            Some(hash) => hash.to_vec(),
+            None => vec![]
+        };
+
+        //Extend the internal key by the commitment data and HashTapTweak it.
+        let mut tweak: Vec<u8> = internal_key.serialize().to_vec();
+        tweak.extend_from_slice(&h);
+        let mut engine = TapTweakHash::engine();
+        engine.input(&tweak);
+        let tweak_value: [u8; 32] = TapTweakHash::from_engine(engine).into_inner();
+            
+
+        //Tweak the internal key by the hashtaptweak
+        let mut taptweaked_key = internal_key.clone();
+        let parity = taptweaked_key.tweak_add_assign(&secp, &tweak_value).expect("Tweak failed");
+        if internal_key.tweak_add_check(&secp, &taptweaked_key, parity, tweak_value) {
+            return Ok(
+                Address {
+                    network: network,
+                    payload: Payload::WitnessProgram {
+                        version: WitnessVersion::V1,
+                        program: taptweaked_key.serialize().to_vec()
+                    }
+                }
+            )
+        } else { panic!("Tweak Failed") } //What error should be used here?
```

Comment:
What error should be returned here if a tweak check fails?

Reply:
I have functions for computing the tweak in currently under review PR #677. That has a function `TapTweakHash::from_key_and_tweak(..)` that can this simpler. 

W.r.t to this error, it is correct to panic here. More discussion on this in https://github.com/rust-bitcoin/rust-bitcoin/pull/677#discussion_r744169773

Reply:
Good to know. Once #677 is complete, I will try and use that function instead.